agent/lib/kontena/actors/container_exec.rb

```diff
@@ -0,0 +1,59 @@
+require_relative '../helpers/rpc_helper'
+
+module Kontena::Actors
+  class ContainerExec
+    include Celluloid
+    include Kontena::Logging
+    include Kontena::Helpers::RpcHelper
+
+    exclusive :input
+
+    attr_reader :uuid
+
+    # @param [Docker::Container] container
+    def initialize(container)
+      @uuid = SecureRandom.uuid
+      @container = container
+      @read_pipe, @write_pipe = IO.pipe
```

Comment:
Not sure if any other naming would make it more clear to which "direction" each of these are meant for. Now, at least for me, it always takes a couple of readings to figure out. :)